### ISSUE TITLE
Improve remove_duplicate_tests.py

### ISSUE THREAD

**Issue by llxia** [adoptium/TKG#395]
With https://github.com/adoptium/TKG/pull/393, we have remove_duplicate_tests.py to remove duplicate tests. However, there are some enhancements that we can make:

- [ ] find a way to persevere comments (whether the license or actual comments within the root element). For example https://stackoverflow.com/questions/33573807/faithfully-preserve-comments-in-parsed-xml. The license can be different from repo to repo. Users should not need to provide `--license-file`. Also, actual comments in the playlist.xml should be preserved.
- [ ] We should compare the content instead of simply removing elements with the duplicated `<testCaseName>`. Only remove if everything under `<test>` is the same. If not, we should print a warning message. 
- [ ] Update remove_duplicate_tests.py to take a git repo. remove_duplicate_tests.py should git clone the repo and check all subdirectories to find playlist.xml and remove duplicate tests if possible.
- [ ] create a GitHub workflow for this task. On a schedule (i.e., biweekly), run remove_duplicate_tests.py against all repos and create auto PR against specific repo if needed.
https://github.com/adoptium/aqa-tests
https://github.com/eclipse-openj9/openj9
https://github.com/adoptium/temurin-build

**Comment by ArkaprabhaChakraborty** [comment]
Hi, can I get this assigned?

**Comment by ArkaprabhaChakraborty** [comment]
PS we might have to make several PRs to this completely issue ina good trackable way :)

**Comment by llxia** [comment]
@ArkaprabhaChakraborty yes, that is correct. This is a top-level issue to track the enhancements. Please divide the deliverables into PRs that are of a reasonable size and contain specific features. Thanks

**Comment by ArkaprabhaChakraborty** [comment]
Where can I add tests for the script? :)

**Comment by ArkaprabhaChakraborty** [comment]
Okay, I tried different ways to parse comments outside the "tree" but none worked with python's default XMLParser... [StackOverflow discussion](https://stackoverflow.com/questions/72253968/preserve-comments-outside-the-root-element-of-xml-file)

A seemingly simple problem just got weirder :)
